net/service/handlers.go

```diff
@@ -0,0 +1,438 @@
+		}
+
+	case pb.Block_COMMENT:
+		return nil, errors.New("TODO")
+	case pb.Block_LIKE:
+		return nil, errors.New("TODO")
+	}
+
+	// handle block
+	return nil, thrd.HandleBlock(signed.Id)
+}
+
+func (s *TextileService) handleFollow(pid peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	if pmes.Payload == nil {
+		return nil, errors.New("payload is nil")
+	}
+	sd := new(pb.SignedData)
+	err := ptypes.UnmarshalAny(pmes.Payload, sd)
+	if err != nil {
+		return nil, err
+	}
+	pubkey, err := libp2pc.UnmarshalPublicKey(sd.SenderPubkey)
+	if err != nil {
+		return nil, err
+	}
+	id, err := peer.IDFromPublicKey(pubkey)
+	if err != nil {
+		return nil, err
+	}
+	data := new(pb.SignedData_Command)
+	err = proto.Unmarshal(sd.SerializedData, data)
+	if err != nil {
+		return nil, err
+	}
+	if data.PeerID != s.node.Identity.Pretty() {
+		return nil, errors.New("follow message doesn't include correct peer id")
+	}
+	if data.Type != pb.Message_FOLLOW {
+		return nil, errors.New("data type is not follow")
+	}
+	good, err := pubkey.Verify(sd.SerializedData, sd.Signature)
+	if err != nil || !good {
+		return nil, errors.New("bad signature")
+	}
+
+	//proof := append(sd.SerializedData, sd.Signature...)
+	//err = s.datastore.Followers().Put(id.Pretty(), proof)
+	//if err != nil {
+	//	return nil, err
+	//}
+	//n := notifications.FollowNotification{notifications.NewID(), "follow", id.Pretty()}
+	//s.broadcast <- n
+	//s.datastore.Notifications().Put(n.ID, n, n.Type, time.Now())
+	log.Debugf("received FOLLOW message from %s", id.Pretty())
+	return nil, nil
+}
+
+func (s *TextileService) handleUnFollow(pid peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	if pmes.Payload == nil {
+		return nil, errors.New("payload is nil")
+	}
+	sd := new(pb.SignedData)
+	err := ptypes.UnmarshalAny(pmes.Payload, sd)
+	if err != nil {
+		return nil, err
+	}
+	pubkey, err := libp2pc.UnmarshalPublicKey(sd.SenderPubkey)
+	if err != nil {
+		return nil, err
+	}
+	id, err := peer.IDFromPublicKey(pubkey)
+	if err != nil {
+		return nil, err
+	}
+	data := new(pb.SignedData_Command)
+	err = proto.Unmarshal(sd.SerializedData, data)
+	if err != nil {
+		return nil, err
+	}
+	if data.PeerID != s.node.Identity.Pretty() {
+		return nil, errors.New("unfollow message doesn't include correct peer id")
+	}
+	if data.Type != pb.Message_UNFOLLOW {
+		return nil, errors.New("data type is not unfollow")
+	}
+	good, err := pubkey.Verify(sd.SerializedData, sd.Signature)
+	if err != nil || !good {
+		return nil, errors.New("bad signature")
+	}
+	//err = s.datastore.Followers().Delete(id.Pretty())
+	//if err != nil {
+	//	return nil, err
+	//}
+	//n := notifications.UnfollowNotification{notifications.NewID(), "unfollow", id.Pretty()}
+	//s.broadcast <- n
+	log.Debugf("received UNFOLLOW message from %s", id.Pretty())
+	return nil, nil
+}
+
+func (s *TextileService) handleOfflineAck(pid peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
```

Comment:
`handleOfflineAck` will be used when the offline handling is in place. in net speak, an "ACK" is an an acknowledgement message. not needed if the receiver is online in our case, since the sender will get a direct response. in offline, the ack is added and retrieved from the DHT... sorta like a text message (offline) vs. a phone call (online)